elements/lisk-genesis/src/create.ts

```diff
@@ -12,30 +12,87 @@
  * Removal or modification of this copyright notice is prohibited.
  */
 
-interface GeneisAccountState {
-	readonly publicKey?: Buffer;
-	readonly address: Buffer;
-	readonly balance: bigint;
-	readonly keys: ReadonlyArray<Buffer>;
-	readonly delegate?: {
-		readonly username: string;
-		readonly lastForgedHeight: bigint;
+import { codec } from '@liskhq/lisk-codec';
+import { hash } from '@liskhq/lisk-cryptography';
+import { LiskValidationError } from '@liskhq/lisk-validator';
+import { EMPTY_BUFFER, EMPTY_HASH } from './constants';
+import {
+	GenesisAccountState,
+	GenesisBlock,
+	GenesisBlockHeaderWithoutId,
+	GenesisBlockParams,
+} from './types';
+import { validateGenesisBlock } from './validate';
+import {
+	genesisBlockHeaderAssetSchema,
+	genesisBlockHeaderSchema,
+} from './schema';
+
+const getBlockId = (header: GenesisBlockHeaderWithoutId): Buffer => {
+	const genesisBlockAssetBuffer = codec.encode(
+		genesisBlockHeaderAssetSchema,
+		header.asset,
+	);
+
+	const genesisBlockHeaderBuffer = codec.encode(genesisBlockHeaderSchema, {
+		...header,
+		...{ asset: genesisBlockAssetBuffer },
+	});
+
+	return hash(genesisBlockHeaderBuffer);
+};
+
+export const createGenesisBlock = (
+	params: GenesisBlockParams,
+): GenesisBlock => {
+	// Default values
+	const initRounds = params.initRounds ?? 3;
+	const height = params.height ?? 0;
+	const timestamp = params.timestamp ?? Math.floor(Date.now() / 1000);
+	const previousBlockID = params.previousBlockID ?? Buffer.from(EMPTY_BUFFER);
+
+	// Constant values
+	const version = 0;
+	const generatorPublicKey = Buffer.from(EMPTY_BUFFER);
+	const reward = BigInt(0);
+	const payload = Buffer.from(EMPTY_BUFFER);
+	const signature = Buffer.from(EMPTY_BUFFER);
+	const transactionRoot = Buffer.from(EMPTY_HASH);
+
+	const { initDelegates } = params;
+
+	const accounts: ReadonlyArray<GenesisAccountState> = [
+		...params.accounts,
```

Comment:
should it consider old address order?

Reply:
I was considering that we will pass new addresses.

Reply:
Add more granular checks afterward along with tests.